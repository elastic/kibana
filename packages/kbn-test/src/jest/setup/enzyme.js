/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * and the Server Side Public License, v 1; you may not use this file except in
 * compliance with, at your election, the Elastic License or the Server Side
 * Public License, v 1.
 */

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
