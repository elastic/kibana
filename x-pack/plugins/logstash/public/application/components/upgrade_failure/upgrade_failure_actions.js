/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiButton, EuiButtonEmpty, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
import { FormattedMessage } from '@kbn/i18n/react';

export function UpgradeFailureActions({ onClose, onRetry, upgradeButtonText }) {
  return (
    <EuiFlexGroup justifyContent="flexStart">
      <EuiFlexItem grow={false}>
        <EuiButton fill onClick={onRetry}>
          {upgradeButtonText}
        </EuiButton>
      </EuiFlexItem>
      <EuiFlexItem grow={false}>
        <EuiButtonEmpty color="primary" onClick={onClose}>
          <FormattedMessage
            id="xpack.logstash.upgradeFailureActions.goBackButtonLabel"
            defaultMessage="Go back"
          />
        </EuiButtonEmpty>
      </EuiFlexItem>
    </EuiFlexGroup>
  );
}

UpgradeFailureActions.propTypes = {
  onClose: PropTypes.func.isRequired,
  onRetry: PropTypes.func.isRequired,
  upgradeButtonText: PropTypes.string.isRequired,
};
