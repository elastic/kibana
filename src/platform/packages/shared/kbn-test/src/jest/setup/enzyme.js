/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the "Elastic License
 * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
 * Public License v 1"; you may not use this file except in compliance with, at
 * your election, the "Elastic License 2.0", the "GNU Affero General Public
 * License v3.0 only", or the "Server Side Public License, v 1".
 */

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

/* eslint-env jest */

/**
 * This is a workaround to fix snapshot serialization of emotion classes when rendering React@18 using `import { render } from 'enzyme'`
 * With React@18 emotion uses `useInsertionEffect` to insert styles into the DOM, which enzyme `render` does not trigger because it is using ReactDomServer.renderToString.
 * With React@17 emotion fell back to sync cb, so it was working with enzyme `render`.
 * Without those styles in DOM the custom snapshot serializer is not able to replace the emotion class names.
 * This workaround ensures a fake emotion style tag is present in the DOM before rendering the component with enzyme making the snapshot serializer work.
 */
function mockEnsureEmotionStyleTag() {
  if (!document.head.querySelector('style[data-emotion]')) {
    const style = document.createElement('style');
    style.setAttribute('data-emotion', 'css');
    document.head.appendChild(style);
  }
}

jest.mock('enzyme', () => {
  const actual = jest.requireActual('enzyme');
  return {
    ...actual,
    render: (node, options) => {
      mockEnsureEmotionStyleTag();
      return actual.render(node, options);
    },
  };
});
