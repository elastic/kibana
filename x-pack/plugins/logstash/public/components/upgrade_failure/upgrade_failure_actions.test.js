/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import { mount, shallow } from 'enzyme';
import { UpgradeFailureActions } from './upgrade_failure_actions';

describe('UpgradeFailureActions component', () => {
  let props;
  let onClose;
  let onRetry;

  beforeEach(() => {
    onClose = jest.fn();
    onRetry = jest.fn();
    props = {
      onClose,
      onRetry,
      upgradeButtonText: 'upgrade button text',
    };
  });

  it('renders component as expected', () => {
    const wrapper = shallow(<UpgradeFailureActions {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onRetry on update click', () => {
    const wrapper = mount(<UpgradeFailureActions {...props} />);
    wrapper.find('EuiButton').simulate('click');
    expect(onRetry).toHaveBeenCalledTimes(1);
  });

  it('calls onClose on "Go back" click', () => {
    const wrapper = mount(<UpgradeFailureActions {...props} />);
    wrapper.find('EuiButtonEmpty').simulate('click');
    expect(onClose).toHaveBeenCalledTimes(1);
  });
});
