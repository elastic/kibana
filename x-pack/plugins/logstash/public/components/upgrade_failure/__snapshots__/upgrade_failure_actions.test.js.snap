// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpgradeFailureActions component renders component as expected 1`] = `
<EuiFlexGroup
  alignItems="stretch"
  component="div"
  direction="row"
  gutterSize="l"
  justifyContent="flexStart"
  responsive={true}
  wrap={false}
>
  <EuiFlexItem
    component="div"
    grow={false}
  >
    <EuiButton
      color="primary"
      fill={true}
      iconSide="left"
      onClick={[MockFunction]}
      type="button"
    >
      upgrade button text
    </EuiButton>
  </EuiFlexItem>
  <EuiFlexItem
    component="div"
    grow={false}
  >
    <EuiButtonEmpty
      color="primary"
      iconSide="left"
      onClick={[MockFunction]}
      type="button"
    >
      <FormattedMessage
        defaultMessage="Go back"
        id="xpack.logstash.upgradeFailureActions.goBackButtonLabel"
        values={Object {}}
      />
    </EuiButtonEmpty>
  </EuiFlexItem>
</EuiFlexGroup>
`;
